lighthouse-core/gather/fetcher.js

```diff
@@ -6,124 +6,56 @@
-  }
-
-  async disable() {
-    if (!this._enabled) return;
-
-    this._enabled = false;
-    await this.session.off('Fetch.requestPaused', this._onRequestPaused);
-    await this.session.sendCommand('Fetch.disable');
-    this._onRequestPausedHandlers.clear();
-  }
-
-  /**
    * @param {string} url
-   * @param {(event: LH.Crdp.Fetch.RequestPausedEvent) => void} handler
-   */
-  async _setOnRequestPausedHandler(url, handler) {
-    this._onRequestPausedHandlers.set(url, handler);
-  }
-
-  /**
-   * @param {LH.Crdp.Fetch.RequestPausedEvent} event
+   * @param {{timeout: number}=} options timeout is in ms
+   * @return {Promise<FetchResponse>}
    */
-  _onRequestPaused(event) {
-    const handler = this._onRequestPausedHandlers.get(event.request.url);
-    if (handler) {
-      handler(event);
-    } else {
-      // Nothing cares about this URL, so continue.
-      this.session.sendCommand('Fetch.continueRequest', {requestId: event.requestId}).catch(err => {
-        log.error('Fetcher', `Failed to continueRequest: ${err.message}`);
-      });
+  async fetchResource(url, options = {timeout: 2_000}) {
+    // In Lightrider, `Network.loadNetworkResource` is not implemented, but fetch
+    // is configured to work for any resource.
+    if (global.isLightrider) {
+      return this._wrapWithTimeout(this._fetchWithFetchApi(url), options.timeout);
     }
-  }
 
-  /**
-   * `Network.loadNetworkResource` was introduced in M88.
-   * The long timeout bug with `IO.read` was fixed in M92:
-   * https://bugs.chromium.org/p/chromium/issues/detail?id=1191757
-   * Lightrider has a bug forcing us to use the old version for now:
-   * https://docs.google.com/document/d/1V-DxgsOFMPxUuFrdGPQpyiCqSljvgNlOqXCtqDtd0b8/edit?usp=sharing&resourcekey=0-aIaIqcHFKG-0dX4MAudBEw
-   * @return {Promise<boolean>}
-   */
-  async shouldUseLegacyFetcher() {
-    const {milestone} = await getBrowserVersion(this.session);
-    return milestone < 92 || Boolean(global.isLightrider);
+    return this._fetchResourceOverProtocol(url, options);
   }
 
   /**
-   * Requires that `fetcher.enable` has been called.
-   *
-   * Fetches any resource in a way that circumvents CORS.
-   *
    * @param {string} url
-   * @param {{timeout: number}=} options timeout is in ms
    * @return {Promise<FetchResponse>}
    */
-  async fetchResource(url, options = {timeout: 2_000}) {
-    if (!this._enabled) {
-      throw new Error('Must call `enable` before using fetchResource');
-    }
+  async _fetchWithFetchApi(url) {
+    // eslint-disable-next-line no-undef
+    const response = await fetch(url);
 
-    if (await this.shouldUseLegacyFetcher()) {
-      return this._fetchResourceIframe(url, options);
-    }
+    let content = null;
+    try {
+      content = await response.text();
+    } catch {}
 
-    return this._fetchResourceOverProtocol(url, options);
+    return {
+      content,
+      status: response.status,
+    };
```

Comment:
iframe fetcher suppressed the response content if the status code was not 2xx. (note: this was not a mistake, even if cached 304 response was received [JS fetch sees](https://github.com/github/fetch/issues/241#issuecomment-161920647) the cached 2xx response code. and by default redirects are followed and the status code is the final response)

Should we consider suppressing the response content here if the status code is not 2xx (like iframe fetcher did)? And also for Network.loadNetworkResponse?

Reply:
Does the browser still work with robots.txt or source maps that have a non-2xx status code?

Reply:
I don't think the status code matters for source maps.

for robots.txt yes it needs to be 2xx https://developers.google.com/search/docs/advanced/robots/robots_txt#http-status-codes

Reply:
I think we should just return anything we get + the status code, and leave it to each use case to check the status if they care. we should be good as it is.

Reply:
> I think we should just return anything we get + the status code, and leave it to each use case to check the status if they care. we should be good as it is.

Yeah SGTM